dds_cli/__main__.py

```diff
@@ -201,117 +195,22 @@ def put(
-                        new_tasks = 0
-
-                        # Get result from future and schedule database update
-                        for fut in done:
-                            uploaded_file = upload_threads.pop(fut)
-                            LOG.debug(
-                                f"Future done for file: {uploaded_file}",
-                            )
-
-                            # Get result
-                            try:
-                                file_uploaded = fut.result()
-                                LOG.info(f"Upload of {uploaded_file} successful: {file_uploaded}")
-                            except concurrent.futures.BrokenExecutor as err:
-                                LOG.critical(
-                                    f"Upload of file {uploaded_file} failed! Error: {err}",
-                                )
-                                continue
-
-                            # Increase the main progress bar
-                            progress.advance(upload_task)
-
-                            # New available threads
-                            new_tasks += 1
-
-                        # Schedule the next set of futures for upload
-                        for next_file in itertools.islice(iterator, new_tasks):
-                            LOG.info(f"Starting: {next_file}")
-                            upload_threads[
-                                texec.submit(
-                                    putter.protect_and_upload,
-                                    file=next_file,
-                                    progress=progress,
-                                )
-                            ] = next_file
-                except KeyboardInterrupt:
-                    LOG.warning(
-                        "KeyboardInterrupt found - shutting down delivery gracefully. "
-                        "This will finish the ongoing uploads. If you want to force "
-                        "shutdown, repeat `Ctrl+C`. This is not advised. "
-                    )
-
-                    # Flag for threads to find
-                    putter.stop_doing = True
-
-                    # Stop and remove main progress bar
-                    progress.remove_task(upload_task)
-
-                    # Stop all tasks that are not currently uploading
-                    _ = [
-                        progress.stop_task(x)
-                        for x in [y.id for y in progress.tasks if y.fields.get("step") != "put"]
-                    ]
-
-        putter.update_project_size()
+    try:
+        dds_cli.data_putter.dds_put(
+            dds_info,
+            dds_info["CONFIG"] if config is None else config,
+            username,
+            project,
+            source,
+            source_path_file,
+            break_on_fail,
+            overwrite,
+            num_threads,
+            silent,
+        )
+    except (dds_cli.exceptions.AuthenticationError, dds_cli.exceptions.UploadError) as e:
```

Comment:
I get the following error: 
```
File "/Users/inaod568/repos/DDS/dds_cli/dds_cli/exceptions.py", line 11, in __str__
    return f"\n:no_entry_sign: {self.message} :no_entry_sign:\n"
AttributeError: 'AuthenticationError' object has no attribute 'message'
```

Don't have time to fix it now though, sorry

Reply:
How did you manage to trigger this? The place it is raised is in the `if self.method != "put":` check and I don't know how to make that fail.. 😅  (so struggling to replicate to fix)